### ISSUE TITLE
winetricks -q still showing popups on kdialog

### ISSUE THREAD

**Issue by Etaash-mathamsetty** [Winetricks/winetricks#1941]
winetricks is still showing most of the messages it normally would even with -q on kdialog, I didn't see this issue on zenity.

**Comment by austin987** [comment]
Winetricks version? I'm going to need instructions to reproduce this..

**Comment by Etaash-mathamsetty** [comment]
> Winetricks version? I'm going to need instructions to reproduce this..

latest winetricks on arch linux
run winetricks -q dotnet48, you will see all the popups show up, but the installer itself runs quietly

**Comment by austin987** [comment]
> > Winetricks version? I'm going to need instructions to reproduce this..
> 
> latest winetricks on arch linux

"latest" isn't a version.

> run winetricks -q dotnet48, you will see all the popups show up, but the installer itself runs quietly no popups on zenity as far I remember

I don't see that in `20220411-next - sha256sum: 2c9a258b3c3f387d40637c76921004dabb8ad41c77ef2c297c50f3c85c56c8c5`, with kwin-5.24.4.

**Comment by Etaash-mathamsetty** [comment]
> > > Winetricks version? I'm going to need instructions to reproduce this..
> > 
> > 
> > latest winetricks on arch linux
> 
> "latest" isn't a version.
> 
> > run winetricks -q dotnet48, you will see all the popups show up, but the installer itself runs quietly no popups on zenity as far I remember
> 
> I don't see that in `20220411-next - sha256sum: 2c9a258b3c3f387d40637c76921004dabb8ad41c77ef2c297c50f3c85c56c8c5`, with kwin-5.24.4.

I see them in:
`20220411 - sha256sum: 69856050cfe9d97902dfd68c1ad42a10a4be0fe76660744126d7e3ed75610495`
kwin-5.25.4

**Comment by austin987** [comment]
I couldn't reproduce..running:
$ winetricks -q dotnet48
in 20220411/current git (20220411-next - sha256sum: 89878ef528995b0e47de4cdff7d551f2ca66e9dfe2631c7b8c63899f5beefdeb)

Then I ran `winetricks -q` and chose `dotnet48` from the menu, and yeah, get GUI popups. Is this what you're considering a bug? Running 'winetricks -q` from a CLI, and expecting the GUI to not give popups?

**Comment by austin987** [comment]
Clarification: you're running `winetricks -q` from the CLI, and THEN choosing what to install (i.e., dotnet48) from the GUI, correct?

**Comment by Etaash-mathamsetty** [comment]
> I couldn't reproduce..running: $ winetricks -q dotnet48 in 20220411/current git (20220411-next - sha256sum: 89878ef528995b0e47de4cdff7d551f2ca66e9dfe2631c7b8c63899f5beefdeb)
> 
> Then I ran `winetricks -q` and chose `dotnet48` from the menu, and yeah, get GUI popups. Is this what you're considering a bug? Running 'winetricks -q` from a CLI, and expecting the GUI to not give popups?

yes

**Comment by austin987** [comment]
I usually wait for someone to send patches for GUI issues, but the notion that zenity/kdialog were behaving differently intrigued me.

Looking into it, zenity was broken and not showing the dialogs at all, when -q was intended to enable timing out the dialog if the user doesn't interact with them. After fixing that regression, I looked into kdialog. Kdialog doesn't have a general --timeout option like zenity, but we can use a different dialog type to support the combination of -q and GUI. I fixed up kdialog for w_info()/w_warn(), by far the most common use cases. For other uses of kdialog though, -q may need more love. Patches welcome.

However, to be clear, -q shouldn't completely suppress dialogs, even in GUI mode. If you don't want to see popups at all, use CLI mode + -q.
